javascript/src/functions.js

```diff
@@ -1,32 +1,41 @@
-const Game = require('./game.js');
+const Game = require("./game.js");
 
-function play(seed) {
+function doRandomRollNumber() {
+  return Math.floor(Math.random() * 6) + 1;
+}
 
-    if (seed != undefined) {
-        require('seedrandom')(seed, { global: true });
-    }
+function wouldAnswerIncorrectly() {
+  return Math.floor(Math.random() * 10) == 7;
+}
 
-    var notAWinner = false;
+function play(seed) {
+  if (seed != undefined) {
+    require("seedrandom")(seed, { global: true });
+  }
 
-    var game = new Game();
+  var notAWinner = false;
 
-    game.add('Chet');
-    game.add('Pat');
-    game.add('Sue');
+  var game = new Game();
 
-    do{
+  game.add("Chet");
+  game.add("Pat");
+  game.add("Sue");
 
-        game.roll(Math.floor(Math.random()*6) + 1);
+  do {
+    game.roll(doRandomRollNumber());
 
-        if(Math.floor(Math.random()*10) == 7){
-            notAWinner = game.wrongAnswer();
-        }else{
-            notAWinner = game.wasCorrectlyAnswered();
-        }
+    if (wouldAnswerIncorrectly()) {
+      notAWinner = game.wrongAnswer();
+    } else {
+      notAWinner = game.wasCorrectlyAnswered();
+    }
+  } while (notAWinner);
```

Comment:
I wanted something like:

```
while (game.canContinue()) {
  game.progress(doRandomRollNumber());
  game.setAnswerState(wouldAnswerCorrectly());
}
```

I think it would better represent a common game loop, but the refactor would have been too much for the time, and would be more of a rewrite than a refactor.